personalizer.py

```diff
@@ -55,16 +54,18 @@ class PersonalizerChain(Chain):
     """
 
     llm_chain: LLMChain
-    workspace: vw.Workspace = None
-    embeddings_model: SentenceTransformer = None
-    next_checkpoint: int = None
+    workspace: Optional[vw.Workspace] = None
+    embeddings_model: SentenceTransformer = SentenceTransformer(
+        "bert-base-nli-mean-tokens"
+    )
+    next_checkpoint: int = 1
     model_save_dir: str = "./"
-    response_checker: SelfResponseChecker = None
+    response_checker: Optional[ResponseChecker]
     check_response: bool = True
```

Comment:
do you consider to remove this and expose response checker initialization to constructor?

Reply:
you can set it from constructor if you want, you can define your own `ResponseChecker` object and pass it in, or do `response_checker=LLMResponseChecker(llm=myllm,prompt=myprompt)`

Reply:
did you mean something else?

Reply:
Nested logic of resolving conflict between check_response and ResponseChecker's existence is a bit confusing.
Maybe just having optional ResponseChecker (None by default, but supplied from constructor) is cleaner?
1) It promotes having definition of actions in rewarding policy next to each other which makes the loop more descriptive
2) It is adding one extra line to existing demo since you need to be more explicit about rewarding policy, but not sure about idea of doing it quietly by default (given that llm call has cost)